kafka-java-auth/src/main/java/io/confluent/kafka/schemaregistry/client/rest/entities/ErrorMessage.java

```diff
@@ -0,0 +1,156 @@
+ * but supports parsing the source JSON error message in more formats
+ * via Jackson annotations (see {@link #unpackNestedError}).
+ *
+ * <p>In additional to the original error format:
+ * <pre>
+ *   {
+ *     "error_code": 400,
+ *     "message": "..."
+ *   }
+ * </pre>
+ * It also supports this format:
+ * <pre>
+ *   {
+ *     "error: {
+ *       "code": 400,
+ *       "message": "..."
+ *     }
+ *   }
+ * </pre>
+ *
+ * Since the class is in the same package
+ * <code>io.confluent.kafka.schemaregistry.client.rest.entities</code>,
+ * pre-pending this library to the classpath will override its implementation
+ * from the Schema Registry Client library and allow parsing Managed Kafka
+ * style error messages.
+ */
+@Schema(description = "Error message")
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ErrorMessage {
+
+  private int errorCode;
+  private String message;
+
+  // Required for Jackson to instantiate the class when error_code/message
+  // fields are not present in the JSON message.
+  public ErrorMessage() {}
+
+  // Required for compatibility with the old class.
+  public ErrorMessage(int errorCode, String message) {
+      this.errorCode = errorCode;
+      this.message = message;
+  }
+
+  @Schema(description = "Error code")
+  @JsonProperty("error_code")
+  public int getErrorCode() {
+    return errorCode;
+  }
+
+  @JsonProperty("error_code")
+  public void setErrorCode(int error_code) {
+    this.errorCode = error_code;
+  }
+
+  @Schema(description = "Detailed error message")
+  @JsonProperty
+  public String getMessage() {
+    return message;
+  }
+
+  @JsonProperty
+  public void setMessage(String message) {
+    this.message = message;
+  }
+
+  @JsonProperty("error")
+  public void unpackNestedError(Map<String, Object> error) {
+    Object code = error.get("code");
+
+    if (code != null) {
+      try {
+        this.errorCode = ((Integer) code);
+      } catch (Exception e1) {
+        try {
+          this.errorCode = Integer.parseInt(code.toString());
+        } catch (Exception e2) {
+          // ignore if can't parse the error code as an integer
+        }
+      }
+    }
+
+    String status = tryGetNonEmptyString(error.get("status"));
+    String message = tryGetNonEmptyString(error.get("message"));
+
+    if (message != null && status != null) {
+      this.message = status + ": " + message;
+    } else if (message != null) {
+      this.message = message;
+    } else if (status != null) {
+      this.message = status;
+    }
+  }
+
+  private static String tryGetNonEmptyString(Object object) {
+    if (object != null) {
+      String s = object.toString();
+      if (s.length() > 0) {
+        return s;
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
```

Comment:
nit: consider instanceof (ErrorMessage)

Reply:
Keeping this as is because that's how it was implemented in the original, to avoid breaking compatibility.